sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracerMetrics.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.LongUpDownCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision;
+
+/**
+ * SDK metrics exported for started and ended spans as defined in the <a
+ * href="https://opentelemetry.io/docs/specs/semconv/otel/sdk-metrics/#span-metrics">semantic
+ * conventions</a>.
+ */
+final class SdkTracerMetrics {
```

Comment:
#nit For alignment with `ExporterInstrumentation`
```suggestion
final class SdkTracerInstrumentation {
```

Reply:
If it's possible, I'd like to suggest the `Metrics` name, and maybe renaming that other one. I think instrumentation is something else, usually using interceptors or such to add telemetry in a somewhat black-box way. These are just shared state + helper methods and don't seem like instrumentation to me (admittedly a personal take).